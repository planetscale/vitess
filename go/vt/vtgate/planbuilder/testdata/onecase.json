[
  {
    "comment": "Add your test case here for debugging and run go test -run=One.",
    "query": "SELECT id, name, (SELECT LENGTH(col) FROM user_extra) AS total_length_col, AVG(id) AS avg_ids FROM user GROUP BY id, name HAVING (SELECT LENGTH(col) FROM user_extra) > 10",
    "plan": {

    }
  }
]