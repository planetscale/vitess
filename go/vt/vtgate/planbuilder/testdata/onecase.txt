# Single table with multi column vindex
"select col1 from foo where (col2,col3) = (1,2)"
{
  "QueryType": "SELECT",
  "Original": "select kid from multicolvin where (column_b, column_c) in (1, 2)",
  "Instructions": {
    "OperatorType": "Route",
    "Variant": "SelectEqualUnique",
    "Keyspace": {
      "Name": "user",
      "Sharded": true
    },
    "FieldQuery": "select kid from multicolvin where 1 != 1",
    "Query": "select kid from multicolvin where (column_b, column_c) in (1, 2)",
    "Table": "multicolvin",
    "Values": [
      1
    ],
    "Vindex": "colb_colc_map"
  }
}
