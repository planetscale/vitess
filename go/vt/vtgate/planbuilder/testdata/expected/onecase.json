[
  {
    "comment": "Add your test case here for debugging and run go test -run=One.",
    "query": "select (select count(*) from user_extra) x from user order by x",
    "plan": {
      "QueryType": "SELECT",
      "Original": "select (select count(*) from user_extra) x from user order by x",
      "Instructions": {
        "OperatorType": "UncorrelatedSubquery",
        "Variant": "PulloutValue",
        "PulloutVars": [
          "__sq1"
        ],
        "Inputs": [
          {
            "InputName": "SubQuery",
            "OperatorType": "Aggregate",
            "Variant": "Scalar",
            "Aggregates": "sum_count_star(0) AS count(*)",
            "Inputs": [
              {
                "OperatorType": "Route",
                "Variant": "Scatter",
                "Keyspace": {
                  "Name": "user",
                  "Sharded": true
                },
                "FieldQuery": "select count(*) from user_extra where 1 != 1",
                "Query": "select count(*) from user_extra",
                "Table": "user_extra"
              }
            ]
          },
          {
            "InputName": "Outer",
            "OperatorType": "UncorrelatedSubquery",
            "Variant": "PulloutValue",
            "PulloutVars": [
              "__sq1"
            ],
            "Inputs": [
              {
                "InputName": "SubQuery",
                "OperatorType": "Aggregate",
                "Variant": "Scalar",
                "Aggregates": "sum_count_star(0) AS count(*)",
                "Inputs": [
                  {
                    "OperatorType": "Route",
                    "Variant": "Scatter",
                    "Keyspace": {
                      "Name": "user",
                      "Sharded": true
                    },
                    "FieldQuery": "select count(*) from user_extra where 1 != 1",
                    "Query": "select count(*) from user_extra",
                    "Table": "user_extra"
                  }
                ]
              },
              {
                "InputName": "Outer",
                "OperatorType": "Route",
                "Variant": "Scatter",
                "Keyspace": {
                  "Name": "user",
                  "Sharded": true
                },
                "FieldQuery": "select :__sq1 as x, __sq1, weight_string(__sq1) from `user` where 1 != 1",
                "OrderBy": "(1|2) ASC",
                "Query": "select :__sq1 as x, __sq1, weight_string(__sq1) from `user` order by __sq1 asc",
                "ResultColumns": 1,
                "Table": "`user`"
              }
            ]
          }
        ]
      },
      "TablesUsed": [
        "user.user",
        "user.user_extra"
      ]
    }
  }
]
